types/foreach/index.d.ts

```diff
@@ -0,0 +1,44 @@
+// Type definitions for foreach 2.0
+// Project: https://github.com/manuelstofer/foreach
+// Definitions by: ExE Boss <https://github.com/ExE-Boss>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Performs the specified action for each element in an array.
+ *
+ * @param callbackfn A function that accepts up to three arguments.
+ *        `forEach` calls the `callbackfn` function one time for each element in the array.
+ * @param thisArg A value to which the `this` keyword can refer in the `callbackfn` function.
+ *        If `thisArg` is omitted, `undefined` is used as the `this` value.
+ */
+declare function forEach<O extends ArrayLike<any>, THIS_ARG = undefined>(
+    array: O,
+    callbackfn: (
+        this: THIS_ARG,
+        value: O extends ArrayLike<infer T> ? T : any,
```

Comment:
```suggestion
declare function forEach<O extends ArrayLike<unknown>, THIS_ARG = undefined>(
    array: O,
    callbackfn: (
        this: THIS_ARG,
        value: O extends ArrayLike<infer T> ? T : unknown,
```
consumers should be forced the deal with the type errors and use any themselves if that's what they want